### PR TITLE
Add ability to read in pddl problem files to plansys2.

### PR DESCRIPTION
**Pull request by xydesa**
Add a new AddProblem service to the plansys2_problem_expert.
Adding problem_file node parameter to plansys2_problem_expert to load a single problem file at launch.

Signed-off-by: Alexander Xydes <alexander.xydes@navy.mil>

### CODE REVIEW

**Comment by fmrico** [comment]
Hi @xydesa 

May you add some tests to this PR? It would be interesting to test what happens when the problem is not found, is not correct, there is errors in the file, or checking that the instances and predicates are correctly loaded.

Thanks!!

**Comment by xydesa** [comment]
@fmrico that's a great idea, I'll start working on that. I'll let you know here when I'm done.

**Comment by xydesa** [comment]
@fmrico I believe I've added tests for all the cases you suggested. I made sure to get coverage over as much of the new code as possible.

**Comment by fmrico** [comment]
Thanks!!!!

File: plansys2_problem_expert/src/plansys2_problem_expert/ProblemExpert.cpp [hunk 1]
```diff
@@ -590,4 +591,103 @@ ProblemExpert::getProblem()
...
+  plansys2_msgs::msg::Tree tree;
+  for (auto ground : problem.init) {
+    auto tree_node = ground->getTree(tree, domain);
+    switch (tree_node->node_type) {
+      case plansys2_msgs::msg::Node::PREDICATE: {
+          plansys2::Predicate pred_node(*tree_node);
+          std::cout << "Adding predicate: " <<
+            parser::pddl::toString(tree, tree_node->node_id) << std::endl;
+          if (!addPredicate(pred_node)) {
+            std::cerr << "Failed to add predicate: " << parser::pddl::toString(
```
[the remark this answers is not in the document]
Comment (fmrico) [reply plansys2_problem_expert/src/plansys2_problem_expert/ProblemExpert.cpp:657]: Yes, I have to check it out soon

**Pull request merged**